tests/typecheck-tests.ts

```diff
@@ -551,6 +561,126 @@ import { ok, err, okAsync, errAsync, fromSafePromise, Result, ResultAsync } from
 
 
 (function describe(_ = 'Combine on Unbounded lists') {
-  // TODO:
-  // https://github.com/supermacro/neverthrow/issues/226
+  (function describe(_ = 'combine') {
+    (function it(_ = 'combines different results into one') {
+      type Expectation = Result<[ number, string, boolean, boolean ], Error | string | string[]>;
+
+      const result: Expectation = Result.combine([
+        ok<number, string>(1),
+        ok<string, string>('string'),
+        err<boolean, string[]>([ 'string', 'string2' ]),
+        err<boolean, Error>(new Error('error content')),
+      ])
+    });
+
+    (function it(_ = 'combines only ok results into one') {
+      type Expectation = Result<[ number, string ], never>;
+
+      const result: Expectation = Result.combine([
+        ok(1),
+        ok('string'),
+      ]);
+    });
+
+    (function it(_ = 'combines only err results into one') {
+      type Expectation = Result<[ never, never ], number | string>;
```

Comment:
### Edge case still not working as expected

When a list of `err` values is passed into any variation of `combine` (whether it's synchronous or asynchronous) **AND** there is no type argument for the success type `T` passed in (thus, the `T` becomes `never`), then we would expect to have a type akin to the one that you have written here on line 586.

**However**, when I try the code out on my machine, the inferred type does not match this expectation:

![image](https://user-images.githubusercontent.com/10621548/196049810-ced1298b-91a9-4d85-81c4-e1b24b18bc6d.png)

The Expectation is correct in this file, but for some reason the type inference logic is not returning what we expect - and for some reason the code still compiles (I don't understand how `never` can be assigned to `[ never, never ]`).

Reply:
@supermacro I have fixed this pretty easily actually, see the line `462` and `432` in `result.ts` file.

I have done this only for the case you have mentioned. If you want to apply this logic to other places as well, simply change `EmptyArrayToNever<...>` to `EmptyArrayToNever<..., 1>`. This extra `1` argument will cause to keep `[never, ...never[]]`-like array as it is. Add `1` here if you want to get precise type such as `[never, never, never]` instead of `never` which was the previous default that would be occurring if all of the elements of the array is `never`.